redash/utils/configuration.py

```diff
@@ -24,6 +24,11 @@ def __init__(self, config, schema=None):
         self.set_schema(schema)
 
     def set_schema(self, schema):
+        if isinstance(schema, dict):
+            for prop in schema.get('properties', {}).values():
+                if 'extendedEnum' in prop:
+                    prop['enum'] = map(lambda v : v['value'], prop['extendedEnum'])
+                    del prop['extendedEnum']
```

Comment:
@arikfr (or @rauchy)

This seemed the best place to put the transformation extendedEnum -> enum logic (so `ConfigurationContainer` stays in sync with what it's expected from jsonschema).

One note is that this modifies the schema in the upper scope (for the DS/Destination handler it didn't seem that big of a deal), so lmk if there is a dict "deep copy" that I should be using in such circumstances.

Reply:
> One note is that this modifies the schema in the upper scope

:thinking: I wonder if we might end up in a situation where it returns the modified version in an API call. Several options:

1. Deep copy.
2. Keep `extendedEnum` and change the UI to use it first and ignore `enum` if `extendedEnum` exists.

Reply:
There is already a preference to `extendedEnum` in UI, just need to keep it there then 😉

Reply:
Deep copy seemed more future-proof, so I went with it in the end :+1: